src/websocket.c

```diff
@@ -79,22 +82,28 @@ kore_websocket_handshake(struct http_request *req, struct kore_wscbs *wscbs)
 	kore_buf_appendf(buf, "%s%s", key, WEBSOCKET_SERVER_RESPONSE);
 	kore_mem_free(key);
 
+#if !defined(KORE_NO_TLS)
 	(void)SHA1_Init(&sctx);
 	(void)SHA1_Update(&sctx, buf->data, buf->offset);
 	(void)SHA1_Final(digest, &sctx);
+#endif
 
 	kore_buf_free(buf);
 
+#if !defined(KORE_NO_TLS)
 	if (!kore_base64_encode(digest, sizeof(digest), &base64)) {
 		kore_debug("failed to base64 encode digest");
 		http_response(req, HTTP_STATUS_INTERNAL_ERROR, NULL, 0);
 		return;
 	}
+#endif
 
 	http_response_header(req, "upgrade", "websocket");
 	http_response_header(req, "connection", "upgrade");
+#if !defined(KORE_NO_TLS)
 	http_response_header(req, "sec-websocket-accept", base64);
```

Comment:
Ok, in order to upgrade a connection to WebSocket, the `Sec-WebSocket-Key` needs to be hashed with SHA1. But the hashing method comes from OpenSSL which we want to exclude. As a result, we need something to replace OpenSSL SHA1 with something not from OpenSSL.